src/components/products/product-card.js

```diff
@@ -1,6 +1,9 @@
 import { useProduct } from "../../context/productList-context";
 import { SortedPrice, ProductRatings, GetFliteredProducts } from "../../filter-utils/filterUtils";
+import { useCart } from "../../context/cart-context";
+import { Link } from "react-router-dom";
 export function GetProducts() {
+  const { cartState: {cart}, cartDispatch } = useCart()
   const { product } = useProduct();
   const getSortedProducts = SortedPrice(product)
   const getRatingProducts = ProductRatings(getSortedProducts)
```

Comment:
this is correct , if possible you can try this with compose function ( if i am wrong please correct me )

const compose = ( ...getfunction ) =>( state , product ) =>getfunction.reduce( ( data , getfunction  ) =>getfunction( state , data ) , product );

const filteredProduct = compose(  SortedPrice , ProductRating , GetFilteredProducts   ) ( state , product  ) 

filteredProduct will contain all filtered products

Reply:
Thank you for the suggestion! It is correct but it is also working fine without the state.